boa/src/object/mod.rs

```diff
@@ -52,10 +52,15 @@ use self::internal_methods::{
 mod tests;
 
 pub(crate) mod internal_methods;
+mod jsarray;
 mod jsobject;
 mod operations;
 mod property_map;
 
+pub use jsarray::*;
+
+pub trait JsObjectType: Into<JsValue> + Into<JsObject> + Deref<Target = JsObject> {}
```

Comment:
Aren't the bounds a bit too restrictive? `Into<JsObject>` already implies `Into<JsValue>`, and with a blanket implementation of `impl<O: JsObjectType> Into<JsValue> for O` we can remove that bound.

The same can be said for `Into<JsObject>`, because `Deref<Target = JsObject>` implies `Into<JsObject>` by `object.deref().clone()`, but in this case I can see some implementors doing some kind of cleanup before cloning the `JsObject`.

Reply:
Hmmm.... The `JsObjectType` trait is an internal thing, that I used to ensure that types implement the required traits, and I should have made it private (`pub(crate)`), And even if we expose it. I think we should seal it.

> `impl<O: JsObjectType> Into<JsValue> for O` 

We should try not implement `Into` trait directly since implementing `From` implicitly implements it. And `impl<O: JsObjectType> From<O> for JsValue` conflicts with `impl<T: Into<JsString>> From<T> for JsValue` in `conversion.rs`

Reply:
Ah, if it's an internal thing then that's okay, we can modify the bounds without breaking the public API :)